CI Pipeline

tlmnrnhrdt opened the pull request.

tlmnrnhrdt commented:
@kpne Ok, this fails because version 1.0.2.720 is not published as a nuget package yet.. what´s the plan here? do we get an updated nuget package or do we need to deploy the `AdSec.1.0.2.720.nupkg` on the github runner?

kpne commented:
@tlmnrnhrdt @peterjgrainger 
The current version of the code is (wrongly) built against an internal version of AdSec_API (1.0.2.72). We are waiting for the next official version to be released (as I understand it, it will be similar to 1.0.3.3 but renamed 1.1.0.0)

nathanrobertsarup reviewed (commented):
Adsec API 1.1.0 is out in nuget
This PR appears to retain OasysUnits.dll (moved location?). Should taht not be from the above nuget?
Various comments inline - happy to discuss

tlmnrnhrdt commented on build-test-deploy.yml:
```diff
@@ -64,5 +68,15 @@ steps:
-  displayName: Deploy yak package 
\ No newline at end of file
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/main'), eq(variables.configuration, 'Release'))
+  displayName: Deploy yak package 
```
We discussed with @peterjgrainger that it is easier if we only have one pipeline that covers both situations. That said the step 'publish nuget package' is still missing in this workflow. Is that what you are referring to?

tlmnrnhrdt commented:
@peterjgrainger @martinfsimpson Hey, would someone be able to finish this setup please?

Missing/not working tasks are currently:
- Check for existing tags
- Package NuGet
- Publishing assets to Github (which might actually work with 'Check for existing tags' working properly)

tlmnrnhrdt commented:
> 

@peterjgrainger Also i used your gitHubConnection for the 'Publishing assets to Github'.. :] Couldn´t figure out how to setup one for myself..

peterjgrainger commented:
> @peterjgrainger @martinfsimpson Hey, would someone be able to finish this setup please?
> 
> Missing/not working tasks are currently:
> 
> * Check for existing tags
> * Package NuGet
> * Publishing assets to Github (which might actually work with 'Check for existing tags' working properly)

@tlmnrnhrdt Do you want the NuGet published to GitHub or Azure DevOps? Azure DevOps is easier to authenticate for people as you authenticate using your arup credentials

tlmnrnhrdt commented:
> > @peterjgrainger @martinfsimpson Hey, would someone be able to finish this setup please?
> > Missing/not working tasks are currently:
> > 
> > * Check for existing tags
> > * Package NuGet
> > * Publishing assets to Github (which might actually work with 'Check for existing tags' working properly)
> 
> @tlmnrnhrdt Do you want the NuGet published to GitHub or Azure DevOps? Azure DevOps is easier to authenticate for people as you authenticate using your arup credentials

@peterjgrainger we only want to 'package' yak and nuget on azure and create a 'pre-release' on github. the actual release of those two packages is another manually triggered github workflow (to be createed).. does that make sense?

peterjgrainger commented:
You want to store as an asset in the release? Or in GitHub packages? You can also make a prerelease in azure by appending the version with -prerelease to avoid it being the newest version and visual studio filters out prerelease by default

tlmnrnhrdt commented:
> You want to store as an asset in the release? Or in GitHub packages? You can also make a prerelease in azure by appending the version with -prerelease to avoid it being the newest version and visual studio filters out prerelease by default

We want it to show up in github > Releases as a 'draft release' (what the task 'Publishing assets to Github' azure does)

tlmnrnhrdt commented:
> > You want to store as an asset in the release? Or in GitHub packages? You can also make a prerelease in azure by appending the version with -prerelease to avoid it being the newest version and visual studio filters out prerelease by default
> 
> We want it to show up in github > Releases as a 'draft release' (what the task 'Publishing assets to Github' azure does)

peterjgrainger commented:
> > > You want to store as an asset in the release? Or in GitHub packages? You can also make a prerelease in azure by appending the version with -prerelease to avoid it being the newest version and visual studio filters out prerelease by default
> > 
> > 
> > We want it to show up in github > Releases as a 'draft release' (what the task 'Publishing assets to Github' azure does)

Got it!

tlmnrnhrdt commented:
Nice, seems to (nearly) work now.

For some reason i struggle to access the variables across different tasks.. do I have to declare them in a different way?

![Capture](https://user-images.githubusercontent.com/45289620/182812622-9be7a100-e1ae-495c-9d78-e361225fb04a.PNG)

Also i don´t think the 'Check for existing tags' is not yet working as expected..

tlmnrnhrdt commented:
@peterjgrainger also @kpne managed to break things again ;)
We changed the platform of the test and converter projects and now nuget packaging doesn´t find our built .dlls any longer?!

tlmnrnhrdt commented:
> @peterjgrainger also @kpne managed to break things again ;) We changed the platform of the test and converter projects and now nuget packaging doesn´t find our built .dlls any longer?!

Ok, fixed again ;)

tlmnrnhrdt commented on build-test-deploy.yml:
```diff
@@ -96,37 +96,77 @@ steps:
+    action: 'edit'
+    target: '$(Build.SourceVersion)'
+    tag: '$(version)'
+    title: 'AdSecGH Version $dotSeparatedTag'
```
@peterjgrainger for some reason this is not picked up here correctly..

peterjgrainger replied:
I'll try to fix this :)

peterjgrainger commented:
@tlmnrnhrdt  I think this is working now. See https://github.com/arup-group/AdSec-Grasshopper/releases/tag/untagged-b84ee00466d36eaed234

tlmnrnhrdt commented on build-test-deploy.yml:
```diff
@@ -153,11 +153,11 @@ steps:
+    publishVstsFeed: '5ac05815-e952-41a1-9899-54474b67ab5d'
+  continueOnError: true
+  condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags'))
+  displayName: Push to NuGet Repository
```
@peterjgrainger the idea is that this would be a separate github workflow.. we only want to do a pre-release on azure and trigger the actual release from github.. 
so i guess that last task should move to a github workflow?!
